lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -576,13 +579,13 @@ fn generate_constraint_init_group(
 
                         // Initialize the token account.
                         let cpi_program = token_program.to_account_info();
-                        let accounts = anchor_spl::token::InitializeAccount3 {
+                        let accounts = anchor_spl::token_2022::InitializeAccount3 {
```

Comment:
Ok, so the idea is that when someone tries to use the helper attributes `token::` to init a new token account it uses the token_2022 account type by default but allows the user to select which token program is used by passing in a different program id for `token_program`

Reply:
Yep that's right. token-2022 is a superset of token, so if you use its instruction bindings and state, you'll be safe for either program.